components/backup/src/endpoint.rs

```diff
@@ -302,38 +353,40 @@ impl BackupRange {
         db: Arc<DB>,
         storage: &LimitedStorage,
         concurrency_manager: ConcurrencyManager,
+        store_id: u64,
         file_name: String,
         backup_ts: TimeStamp,
         start_ts: TimeStamp,
         compression_type: Option<SstCompressionType>,
         compression_level: i32,
+        region_max_size: u64,
+        region: Region,
     ) -> Result<(Vec<File>, Statistics)> {
-        let mut writer = match BackupWriter::new(
+        let writer = match BackupWriter::new(
             db,
+            store_id,
             &file_name,
             storage.limiter.clone(),
             compression_type,
             compression_level,
+            region_max_size,
+            region,
```

Comment:
It accepts too many arguments, could refactor into builder pattern? https://doc.rust-lang.org/1.0.0/style/ownership/builders.html